src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Utility/GetPackageReferenceUtility.cs

```diff
@@ -58,6 +58,27 @@ internal static PackageIdentity UpdateResolvedVersion(LibraryDependency projectL
             return new PackageIdentity(projectLibrary.Name, resolvedVersion);
         }
 
+        /// <summary>
+        /// Gets the version for the transitive package and target framework.
+        /// </summary>
+        /// <param name="dependencyLibrary">Library from the assets file.</param>
+        /// <param name="targetFramework">Target framework from the project file.</param>
+        /// <param name="targets">Target assets file with the package information.</param>
+        internal static PackageIdentity GetResolvedVersionTransitive(PackageDependency dependencyLibrary, NuGetFramework targetFramework, IEnumerable<LockFileTarget> targets)
```

Comment:
The method body doesn't seem to be doing anything that's specific to transitive packages. I think the same should work for direct dependencies. I see that `GetInstalledVersion` below is slightly different, it uses a `TargetFrameworkInformation` for something, but I don't really understand what/why. I think if it's deleted it might work the same, be more efficient, and work the same as your implementation. Maybe they could be merged into a single implementation?

Reply:
Combined and simplified these to use GetInstalledVersion.